### ISSUE TITLE
[data grid] `Cmd+Enter` doesn't toggle detail panel on macOS

### ISSUE THREAD

**Issue by cherniavskii** [mui/mui-x#4388]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

`Cmd+Enter` shortcut on macOS doesn't toggle detail panel

### Expected behavior 🤔

As per https://mui.com/components/data-grid/accessibility/#group-amp-pivot, `Ctrl+Enter` should toggle detail panel.
On macOS `Ctrl` should be replaced by `⌘ Command` key: (see https://mui.com/components/data-grid/accessibility/#key-assignment-conventions), therefore `Cmd+Enter` should toggle detail panel

### Steps to reproduce 🕹

Steps:

1. Go to https://mui.com/components/data-grid/group-pivot/#master-detail on macOS
2. Click on any cell in master/detail demo
3. Press `Cmd+Enter`

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>
